src/raftstore/store/worker/read.rs

```diff
@@ -150,110 +149,31 @@ impl Progress {
     }
 }
 
-pub enum Task {
-    Register(ReadDelegate),
-    Update((u64, Progress)),
-    Read(RaftCommand),
-    Destroy(u64),
-}
-
-impl Task {
-    pub fn register(peer: &Peer) -> Task {
-        let delegate = ReadDelegate::from_peer(peer);
-        Task::Register(delegate)
-    }
-
-    pub fn update(region_id: u64, progress: Progress) -> Task {
-        Task::Update((region_id, progress))
-    }
-
-    pub fn destroy(region_id: u64) -> Task {
-        Task::Destroy(region_id)
-    }
-
-    #[inline]
-    pub fn read(cmd: RaftCommand) -> Task {
-        Task::Read(cmd)
-    }
-
-    /// Task accepts `RaftCmdRequest`s that contain Get/Snap requests.
-    /// Returns `true`, it can be saftly sent to localreader,
-    /// Returns `false`, it must not be sent to localreader.
-    #[inline]
-    pub fn acceptable(request: &RaftCmdRequest) -> bool {
-        if request.has_admin_request() || request.has_status_request() {
-            false
-        } else {
-            for r in request.get_requests() {
-                match r.get_cmd_type() {
-                    CmdType::Get | CmdType::Snap => (),
-                    CmdType::Delete
-                    | CmdType::Put
-                    | CmdType::DeleteRange
-                    | CmdType::Prewrite
-                    | CmdType::IngestSST
-                    | CmdType::Invalid => return false,
-                }
-            }
-            true
-        }
-    }
-}
-
-impl Display for Task {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        match *self {
-            Task::Register(ref delegate) => write!(f, "localreader Task::Register {:?}", delegate),
-            Task::Read(ref cmd) => write!(f, "localreader Task::Read {:?}", cmd.request),
-            Task::Update(ref progress) => write!(f, "localreader Task::Update {:?}", progress),
-            Task::Destroy(region_id) => write!(f, "localreader Task::Destroy region {}", region_id),
-        }
-    }
-}
-
+#[derive(Clone)]
 pub struct LocalReader<C: ProposalRouter> {
-    store_id: u64,
+    store_id: Option<u64>,
```

Comment:
why using Option here?

Reply:
The LocalReader object needs to be created before getting the store_id(node.rs: L119), so I use an Option for initializing the store_id later(read.rs: L221)